### PR TITLE
Skip all-false nulls array in MapBlockBuilder

### PR DESCRIPTION
**Pull request by raunaqmorarka**
## Description

Skip all-false nulls array in MapBlockBuilder

> Is this change a fix, improvement, new feature, refactoring, or other?

improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

core query engine

> How would you describe this change to a non-technical end user or system administrator?

minor performance improvement to operations on MAP data type

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: core/trino-spi/src/main/java/io/trino/spi/block/MapBlockBuilder.java [hunk 1]
```diff
@@ -279,7 +288,7 @@ public Block build()
                 getMapType(),
                 0,
                 positionCount,
-                Optional.of(mapIsNull),
+                hasNullValue ? Optional.of(mapIsNull) : Optional.empty(),
```
Comment (lukasz-stec) [inline core/trino-spi/src/main/java/io/trino/spi/block/MapBlockBuilder.java:291]: How about adding a unit test for this?

**Review by lukasz-stec — APPROVED** [review]
lgtm % test

File: core/trino-spi/src/main/java/io/trino/spi/block/MapBlockBuilder.java (hunk 1 above)
Comment (sopel39) [reply core/trino-spi/src/main/java/io/trino/spi/block/MapBlockBuilder.java:291]: Good idea, there is `TestMapBlock`

**Review by sopel39 — APPROVED** [review]

File: core/trino-spi/src/main/java/io/trino/spi/block/MapBlockBuilder.java (hunk 1 above)
Comment (raunaqmorarka) [reply core/trino-spi/src/main/java/io/trino/spi/block/MapBlockBuilder.java:291]: added tests

**Pull request merged**